src/samples/audio/detect_test.c

```diff
@@ -140,6 +150,31 @@ static void notify_host(const struct comp_dev *dev)
 #endif /* CONFIG_IPC_MAJOR_4 */
 }
 
+#if CONFIG_AMS
+
+/* KPD message*/
+DEFINE_UUID(ams_kpd_msg_uuid, AMS_KPD_MSG_UUID);
+
+static int ams_notify_kpb(const struct comp_dev *dev)
+{
+	struct comp_data *cd = comp_get_drvdata(dev);
+	struct ams_message_payload ams_payload;
+
+	cd->client_data.r_ptr = NULL;
+	cd->client_data.sink = NULL;
+	cd->client_data.id = 0; /**< TODO: acquire proper id from kpb */
+	/* time in milliseconds */
```

Comment:
I don't see a variable name that looks like a time or duration, is this comment intentional?

````c
uint32_t drain_req; /** defines draining size in bytes. */
````

Reply:
@plbossart yes, this comment was intentional, `drain_req `should be in milliseconds. Maybe with previous implementations  of KPB module draining size was in bytes but now KPB accepts draining size in ms. In `kpb_init_drainig()` it then calculates the number of bytes based on `drain_req ` value received in notification payload.
`size_t drain_req = cli->drain_req * kpb->config.channels *
			       (kpb->config.sampling_freq / 1000) *
			       (KPB_SAMPLE_CONTAINER_SIZE(sample_width) / 8);`